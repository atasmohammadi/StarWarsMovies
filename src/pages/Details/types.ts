import type { Movie } from '../Home/types';

export interface DetailsScreenPropsType {
  route: {
      params: {
          movie: Movie
      }
  }
}
