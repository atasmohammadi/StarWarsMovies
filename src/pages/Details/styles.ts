import {StatusBar, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: StatusBar.currentHeight || 0,
  },
  item: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#fff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
    borderRadius: 5,
  },
  row: {
    flexDirection: 'row',
  },
  title: {
    fontSize: 14,
    fontWeight: 'bold',
  },
  desc: {
    fontSize: 14,
  },
  imageContainer: {
    flex: 3,
    alignItems: 'center',
    justifyContent: 'center',
  },
  descriptionContainer: {
    flex: 1,
    padding: 20,
  },
  itemImage: {
    width: '100%',
    height: '100%',
  },
});

export default styles;
