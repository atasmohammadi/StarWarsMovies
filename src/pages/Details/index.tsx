import * as React from 'react';
import { SafeAreaView, View, Text, Image } from 'react-native';
import styles from './styles';
import type { DetailsScreenPropsType } from './types';

export default function Details(props: DetailsScreenPropsType): React.ReactNode {
  const { route } = props;
  const { movie } = route.params;

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.item}>
        <View style={styles.imageContainer}>
          {movie.Poster && <Image
            style={styles.itemImage}
            source={{
              uri: movie.Poster,
            }}
          />}
        </View>
        <View style={styles.descriptionContainer}>
          {movie.title && (
            <View style={styles.row}>
              <Text style={styles.title}>Title: </Text>
              <Text style={styles.desc}>{movie.title}</Text>
            </View>
          )}
          {movie.release_date && (
            <View style={styles.row}>
              <Text style={styles.title}>Release Date: </Text>
              <Text style={styles.desc}>{movie.release_date}</Text>
            </View>
          )}
          {movie.characters && (
            <View style={styles.row}>
              <Text style={styles.title}>Characters: </Text>
              <Text style={styles.desc}>{movie.characters}</Text>
            </View>
          )}
        </View>
      </View>
    </SafeAreaView>
  );
}
