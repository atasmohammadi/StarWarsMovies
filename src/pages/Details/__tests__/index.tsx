import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../index';

const mockedProps = {
  route: {
    params: {
      movie: {
        imdbID: 'id2',
        Title: 'movie2',
        Poster: 'url',
        Year: '2001',
        Type: 'movie',
        page: 1,
      }
    }
  }
};

describe('Details Screen', () => {
  let component: React.ReactNode;
 
  beforeEach(() => {
    component = renderer.create(
      <Details {...mockedProps} />
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});