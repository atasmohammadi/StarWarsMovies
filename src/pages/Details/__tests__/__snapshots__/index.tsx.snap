// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Screen should render with given state from Redux store 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "elevation": 3,
        "flex": 1,
        "flexDirection": "column",
        "marginHorizontal": 16,
        "marginVertical": 8,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.22,
        "shadowRadius": 2.22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "url",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Title: 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          movie2
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Year: 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          2001
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Type: 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          movie
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
